### ISSUE TITLE
Get 404 if manually enter subpath when on default language

### ISSUE THREAD

**Issue by mhaidar** [isaachinman/next-i18next#469]
## Describe the bug
If I visit https://myURL and my browser language is en-US (which is the defaultLanguage)
If I try to go to https://myURL/en-AU, I will get a 404

If I visit https://myURL and my browser language is pl, the URL will automatically changed to https://myURL/pl.
If I try to go to https://myURL/en-AU, It will work.

This is a change of behaviour from 0.40.1 when I was always able to manually change the sub path.

## Occurs in next-i18next version
v1.1.0

## Steps to reproduce
Refer to description

## Expected behaviour
Refer to description

## Screenshots
If applicable, add screenshots or a GIF to help explain your problem.

## OS (please complete the following information)
 - Device: MacOS Mojave 10.14.6
 - Browser: Chrome Version 76.0.3809.100

## Additional context
My settings:
```
  "defaultLanguage": "en-US",
  "otherLanguages": [
    "en",
    "en-AU",
    "es",
    "pl",
    "ar",
    "zh-Hant"
  ],
  "localeSubpaths": {
    "en-AU": "au",
    "en": "en",
    "es": "es",
    "pl": "pl",
    "ar": "ar",
    "zh-Hant": "zh"
  },
  "fallbackLng": "en",
  "defaultNS": "messages"
```

**Comment by isaachinman** [comment]
Based on the `localeSubpaths` value you've provided, `/eu-AU` is not a valid route.

**Comment by mhaidar** [comment]
Sorry my example was not accurate.
To clarify, when I hit the base URL and I am on the default language, the base URL stays the same.
If I then modify my URL to e.g baseURL/pl, it will not switch the language, it will give a 404.

**Comment by StarpTech** [comment]
Hi, I have the same issue but regardless what my start url is. As soon as I enter a different locale I get an error.

e.g

1. go to localhost (default)
2. enter localhost/en
3. The redirect end up in 404 and url `http://localhost:8082/de/en`

**Comment by StarpTech** [comment]
I could fix it by providing all `localSubPaths`
```
{
  localeSubpaths: {
    de: 'de',
    en: 'en'
  }
}
```
this works because an regex is [generated](https://github.com/isaachinman/next-i18next/blob/master/src/middlewares/next-i18next-middleware.ts#L22) based on these values but after this change my default language doesn't work it will always redirect to the url with the appended locale.

**Comment by StarpTech** [comment]
@mhaidar as asked [here](https://github.com/isaachinman/next-i18next/issues/472#issuecomment-525488117) that behaviour you describe is intentional. There is no "default" path without a locale but the user is redirected to the proper locale.

**Comment by isaachinman** [comment]
The new behaviour can be described as such:

1. Routes without a locale subpath go through the normal i18next language detection
2. Routes with a locale subpath are enforced to that locale

**Comment by mhaidar** [comment]
@StarpTech I did manage to access `localhost:8082/` using the default language. 
My problem is I could not switch the language manually in the web browser path. 
Taking a closer look, I don't think it is related to the default language. If I land on the URL that has no language appended, then I can not switch the language. 

Try this with your browser language set to English UK.

```
{
  defaultLanguage: 'en'
  otherLanguages: ['de', 'en-UK']
  localeSubpaths: {
    de: 'de'
  }
}
```

You will land on `localhost:8082/` and the correct en-UK translation.
Then try changing to `localhost:8082/de`, you will get a 404

**Comment by isaachinman** [comment]
This looks like a slightly nuanced bug. The use of [`subpathIsRequired` here](https://github.com/isaachinman/next-i18next/blob/master/src/middlewares/next-i18next-middleware.ts#L49) is returning `false` when you are entering a subpath route for a locale which does not actually require a subpath itself.

We'll need to shift some of the logic around and cover this with tests.

Let me know if either if you are willing to undertake that work - I'm very busy with commercial work at the moment.

**Comment by StarpTech** [comment]
@mhaidar I can reproduce it.

**Comment by isaachinman** [comment]
Yes, it's fully reproducible and I explained the root cause above. Contributions welcome!

**Comment by StarpTech** [comment]
@isaachinman please add "bug" label

**Comment by nuanyang233** [comment]
I guess the bug is caused by the routing of 'nextjs'

**Comment by StarpTech** [comment]
@nuanyang233 no look at https://github.com/isaachinman/next-i18next/issues/469#issuecomment-526819867 it's an unhandled case in this lib.

**Comment by JohannesHome** [comment]
I also have a bug with a 404 that is pretty severe, I'm not sure if this is a separate issue though. 

It occurs when not using english as as default.

When I start the example with this configuration: (with `de` as the default instead of `en`)
```
{
  defaultLanguage: 'de',
  localeSubpaths: {
    en: 'en',
  },
  otherLanguages: ['en'],
}
```

and I go to `http://localhost:3000/en/second-page` with a `de` browser the first time (no cookies) I get a 404. 

Am I maybe missing a configuration here?

**Comment by isaachinman** [comment]
@JohannesHome Sounds like a separate bug. Please open a new issue. As always, bugs will get fixed faster if people contribute fixes.

**Comment by isaachinman** [comment]
Hi there - could those users experiencing this problem try testing their app against #498?

**Comment by JohannesHome** [comment]
@isaachinman works for me.

**Comment by isaachinman** [comment]
This has been released in v1.2.0.

**Comment by mhaidar** [comment]
Works, thank you @isaachinman
